lib/compat/wordpress-6.2/get-global-styles-and-settings.php

```diff
@@ -189,3 +189,38 @@ function _gutenberg_get_global_stylesheet_clean_cache_upon_upgrading( $upgrader,
 	}
 }
 
+/*
+ * Function to get the settings resulting of merging core, theme, and user data.
+ *
+ * @param array $path    Path to the specific setting to retrieve. Optional.
+ *                       If empty, will return all settings.
+ * @param array $context {
+ *     Metadata to know where to retrieve the $path from. Optional.
+ *
+ *     @type string $block_name Which block to retrieve the settings from.
+ *                              If empty, it'll return the settings for the global context.
+ *     @type string $origin     Which origin to take data from.
+ *                              Valid values are 'all' (core, theme, and user) or 'base' (core and theme).
+ *                              If empty or unknown, 'all' is used.
+ * }
+ *
+ * @return array The settings to retrieve.
+ */
+function gutenberg_get_global_settings( $path = array(), $context = array() ) {
+	if ( ! empty( $context['block_name'] ) ) {
+		$path = array_merge( array( 'blocks', $context['block_name'] ), $path );
```

Comment:
Let's not use `array_merge` and rather combine the arrays via iterating over `$path`. See https://github.com/WordPress/wordpress-develop/pull/3555.

Reply:
Addressed https://github.com/WordPress/gutenberg/pull/45971/commits/178cf4e44f5eb515ca0f845cbf706592c075b8c1